cmd/mattermost/commands/team.go

```diff
@@ -285,6 +302,54 @@ func searchTeamCmdF(command *cobra.Command, args []string) error {
 	return nil
 }
 
+func archiveTeamsCmdF(command *cobra.Command, args []string) error {
+	a, err := InitDBCommandContextCobra(command)
+	if err != nil {
+		return err
+	}
+	defer a.Shutdown()
+
+	if len(args) < 1 {
+		return errors.New("Enter at least one team to archive.")
+	}
+	teams := getTeamsFromTeamArgs(a, args)
+	for i, team := range teams {
+		if team == nil {
+			CommandPrintErrorln("Unable to find team '" + args[i] + "'")
+			continue
+		}
+		if result := <-a.Srv.Store.Team().Delete(team.Id, model.GetMillis()); result.Err != nil {
+			CommandPrintErrorln("Unable to archive team '" + team.Name + "' error: " + result.Err.Error())
+		}
+	}
+
+	return nil
+}
+
+func restoreTeamsCmdF(command *cobra.Command, args []string) error {
+	a, err := InitDBCommandContextCobra(command)
+	if err != nil {
+		return err
+	}
+	defer a.Shutdown()
+
+	if len(args) < 1 {
+		return errors.New("Enter at least one team.")
+	}
+
+	teams := getTeamsFromTeamArgs(a, args)
+	for i, team := range teams {
+		if team == nil {
+			CommandPrintErrorln("Unable to find team '" + args[i] + "'")
+			continue
+		}
+		if result := <-a.Srv.Store.Team().SetDeleteAt(team.Id, 0, model.GetMillis()); result.Err != nil {
```

Comment:
Normally we use the "App layer" (`app/team.go` in this case) for implement the commands, we don't call directly the store.